File: src/test/java/com/automation/pages/AmazonHomePage.java
package com.automation.pages;

import com.automation.utils.WaitUtils;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class AmazonHomePage {

    private final WebDriver driver;

    private final By searchTextbox = By.id("twotabsearchtextbox");
    private final By searchSubmitButton = By.id("nav-search-submit-button");

    public AmazonHomePage(WebDriver driver) {
        this.driver = driver;
    }

    public void searchProduct(String productName) {
        WaitUtils.waitForVisibility(driver, searchTextbox).clear();
        WaitUtils.waitForVisibility(driver, searchTextbox).sendKeys(productName);
        WaitUtils.waitForClickable(driver, searchSubmitButton).click();
    }
}

File: playwright-ts/pages/amazonHome.page.ts
import { Locator, Page } from '@playwright/test';

export class AmazonHomePage {
  readonly page: Page;
  readonly searchTextbox: Locator;
  readonly searchSubmitButton: Locator;

  constructor(page: Page) {
    this.page = page;
    this.searchTextbox = page.locator('#twotabsearchtextbox');
    this.searchSubmitButton = page.locator('#nav-search-submit-button');
  }

  async goto(): Promise<void> {
    await this.page.goto('/');
  }

  async searchProduct(productName: string): Promise<void> {
    await this.searchTextbox.fill(productName);
    await this.searchSubmitButton.click();
  }
}
